src/lib.rs

```diff
@@ -335,20 +425,32 @@ fn generate(synapse_data_directory: &Path) -> Result<(), Error> {
     Ok(())
 }
 
-/// Raise an image.
-fn up_image(synapse_data_directory: &Path, create_new_container: bool) -> Result<(), Error> {
+/// Ensures that a container is running for the synapse image.
+/// If a network is provided then the container will be given the hostname `synapse`.
+fn up_image(
+    synapse_data_directory: &Path,
+    container_config: &ContainerConfig,
+    create_new_container: bool,
+) -> Result<(), Error> {
     let container_name = "mx-tester_synapse";
-    let container_up = is_container_up(container_name);
+    let container_up = is_container_up(container_name)?;
     if container_up && create_new_container {
         container_stop(container_name)
     } else if container_up {
         return Ok(());
     }
-    if is_container_built(container_name) {
-        container_rm(container_name)
+    if is_container_built(container_name)? {
+        container_rm(container_name)?;
     }
     let mut command = std::process::Command::new("docker");
     command.arg("run");
+    if container_config.docker_network.is_some() {
```

Comment:
```rust
if let Some(ref network) = container_config.docker_network {
   ...
   .arg(network)
   ...
}
```

This makes clear to both developer and compiler that the `unwrap()` below cannot fail.